keps/sig-auth/4601-authorize-with-selectors/README.md

```diff
@@ -0,0 +1,795 @@
+	// values is an array of string values. If the operator is In or NotIn,
+	// the values array must be non-empty. If the operator is Exists or DoesNotExist,
+	// the values array must be empty. This array is replaced during a strategic
+	// merge patch.
+	// +optional
+	// +listType=atomic
+	Values []string `json:"values,omitempty" protobuf:"bytes,3,rep,name=values"`
+}
+
+
+```
+
+Importantly, if old webhook authorizers do not honor these new field, they will assume the broadest possible access and fail closed.
+If old in-cluster authorization does not include field and label selectors, the kube-apiserver will assume the broadest possible access and fail closed.
+
+### User Stories (Optional)
+
+#### As a SAR client, I want to check a request with a field or label selector
+
+This type of usage probably finds the stringified serialization format used in the query parameters the
+most convenient format to build their request with.
+Providing the query parameter serialization format avoids the need for a client to grow a decently complex lexer/parser.
+
+#### As an authorization webhook author, I want to easily consume the field and label selectors
+
+This type of usage probably finds a serialized `[]Requirement` to be the most convenient way to consume the field and label selector.
+Providing the parsed value avoids the need for every consumer to grow a decently complex lexer/parser.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+#### client provides field or label selector to kube-apiserver that does not parse
+
+The kube-apiserver does not parse the selectors until requested.
+This means that if the authorizers somehow use string matching, they may use the rawSelector values to authorize.
+It will be up to the REST handler to accept or reject requests for bad selectors.
+This approach allows an aggregated API server to have extended field and label selector syntax, though we strongly discourage doing so.
+If the authorizer uses the parsed selectors (far more likely), the authorizer should make an explicit choice about what to do.
+Options include
+1. Error on unparseable selector and NoOpinion (continue to next authorizer).
+2. Error on unparseable selector and Fail (do not continue to next authorizer, we know we should have authorized)
+3. Attempt to authorize without the selector information.
+   If the client is authorized without the selector, then Allow since they have broader permission.
+   If the client is not authorized without the selector then either NoOpinion or Fail depending on intent.
+
+I suggest option #3 as generally the best.
+
+#### client provides field or label selector to kube-apiserver with improper verb
```

Comment:
what about a SAR that includes selector requirements and a non-list/watch/deletecollection verb?